apps/devhub/tasks.py

```diff
@@ -67,16 +68,29 @@ def validate_and_submit(addon, file_, listed=None):
 def submit_file(addon_pk, file_pk):
     addon = Addon.unfiltered.get(pk=addon_pk)
     file_ = FileUpload.objects.get(pk=file_pk)
-    if (file_.passed_all_validations and
-            not addon.fileupload_set.filter(
-                created__gt=file_.created, version=file_.version).exists()):
+    if file_.passed_all_validations:
+        create_version_for_upload(addon, file_)
+    else:
+        log.info('Skipping version creation for {file_id} that failed '
+                 'validation'.format(file_id=file_pk))
+
+
+@write
+@transaction.atomic
+def create_version_for_upload(addon, file_, sleep=None):
+    if (addon.fileupload_set.filter(
+            created__gt=file_.created, version=file_.version).exists()
```

Comment:
I agree. Interestingly my PEP8 linter complains about that. I switched it to line up with the `(`.